dlt/normalize/items_normalizers.py

```diff
@@ -52,6 +63,120 @@ def __init__(
+        returns a single TSchemaUpdate if a new column was added, otherwise None.
+        """
+        schema = self.schema
+        schema_update: TSchemaUpdate = {}
+
+        column_replaced = False
+        for i, select in enumerate(parsed_select.selects):
+            # Check if the column already exists
+            select_alias = (
+                select.alias_or_name.lower()
+                if isinstance(select, sqlglot.exp.Alias)
+                else (
+                    select.output_name.lower() if isinstance(select, sqlglot.exp.Column) else None
+                )
+            )
+
+            if select_alias == column_name:
+                # Replace the existing column
+                parsed_select.selects[i] = alias_expression
+                column_replaced = True
+                break
+
+        if not column_replaced:
+            # Append the column if it doesn't exist
+            parsed_select.selects.append(alias_expression)
+
+            partial_table = normalize_table_identifiers(
+                {
+                    "name": root_table_name,
+                    "columns": {
+                        column_name: (
+                            dlt_id_column() if column_name == "_dlt_id" else dlt_load_id_column()
+                        )
+                    },
+                },
+                schema.naming,
+            )
+
+            table_updates = schema_update.setdefault(root_table_name, [])
+            table_updates.append(partial_table)
+
+            return schema_update
+
+        return None
+
+    def __call__(self, extracted_items_file: str, root_table_name: str) -> List[TSchemaUpdate]:
+        with self.normalize_storage.extracted_packages.storage.open_file(
+            extracted_items_file, "r"
+        ) as f:
+            select_dialect, select_statement = read_dialect_and_sql(
+                file_obj=f,
+                fallback_dialect=self.config.destination_capabilities.sqlglot_dialect,  # caps are available at this point
+            )
+
+        parsed_select = sqlglot.parse_one(select_statement, read=select_dialect)
+        parsed_select = cast(sqlglot.exp.Select, parsed_select)
+
+        schema_updates = []
+
+        if self.config.json_normalizer.add_dlt_load_id:
+            dlt_load_id_alias = sqlglot.exp.Alias(
+                this=sqlglot.exp.Literal.string(self.load_id),
+                alias=sqlglot.exp.to_identifier("_dlt_load_id"),
+            )
+            load_id_update = self._add_or_replace_column(
+                parsed_select, "_dlt_load_id", dlt_load_id_alias, root_table_name
+            )
+            if load_id_update:
+                schema_updates.append(load_id_update)
+
+        if (
+            self.config.json_normalizer.add_dlt_id
+            and self.config.destination_capabilities.sqlglot_dialect != "redshift"
+        ):
+            # TODO: redshift doesn't have an in-built function that generates unique values
```

Comment:
They have these user defined functions in redshift, where you essentially can just use a python method. I'm not sure we will be able to get this to work with sqlglot though. If it does not, we should raise here in case adding dlt_id is set to true.

Reply:
Another idea, that I maybe mentioned elsewhere: you may be able to get the row_number of the currently processing row and combine it with a hashfunction (redshift has both features afaik) to get a mechanism to create a unique id.